timelock-server/src/integ2Test/java/com/palantir/atlasdb/timelock/AbstractMultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -262,33 +263,30 @@ public void canCreateNewClientsDynamically() {
     }
 
     @Test
-    @Disabled // TODO(boyoruk): Investigate why this test fails
     public void lockRequestCanBlockForTheFullTimeout() {
         LockToken token =
                 client.lock(LockRequest.of(LOCKS, DEFAULT_LOCK_TIMEOUT_MS)).getToken();
 
-        try {
-            LockResponse response = client.lock(LockRequest.of(LOCKS, LONGER_THAN_READ_TIMEOUT_LOCK_TIMEOUT_MS));
-            assertThat(response.wasSuccessful()).isFalse();
-        } finally {
-            client.unlock(token);
-        }
+        Assertions.assertThatThrownBy(() -> {
+                    client.lock(LockRequest.of(LOCKS, LONGER_THAN_READ_TIMEOUT_LOCK_TIMEOUT_MS));
+                })
+                .hasCause(READ_TIMED_OUT_EXCEPTION);
+
+        client.unlock(token);
```

Comment:
There is a logic error in this test. To verify that lock requests can block for the full read timeout, we should expect a read timeout exception at the end. The sequence of events should be as follows:

1. The request is blocked until the first read timeout occurs.
2. After the first read timeout, since `acquireTimeoutMs` is longer than the read timeout, the request should be attempted a second time.
3. After the second attempt, since the deadline has passed, a read timeout exception should be raised.
4. Finally, by observing the read timeout exception, we confirm that the lock request can indeed block for the full timeout.

Reply:
The original test used `finally` and I think this was to try and unlock the token even if the assertion fails, so we should try to keep that behaviour (the idea being that this could cause other tests that might try to acquire the same lock to also fail even when they shouldn't, making it more difficult to isolate the problem if we added a bug here). 

That said, the original test also does not quite do that correctly: have a look at `multipleLockRequestsWithTheSameIdAreGranted` and see if we can adapt this here?